File: wiki_starwars/lib/model/planet_model.dart
class PlanetModel {
  String _name;

  PlanetModel({String name});

  String get name => _name;
  set name(String name) => _name = name;

  PlanetModel.toMap(Map<String, dynamic> json) {
    _name = json['name'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['name'] = this._name;
    return data;
  }
}

File: app/src/main/java/com/siedg/desafio_android/data/model/PlanetModel.kt
package com.siedg.desafio_android.data.model

import com.google.gson.annotations.SerializedName

class PlanetModel (
    val climate: String,
    val diameter: String,
    val gravity: String,
    val name: String,
    val orbitalPeriod: String,
    val population: String,
    val residents: List<String>,
    val rotationPeriod: String,
    val surfaceWater: String,
    val terrain: String,
    val url: String
)